### PR TITLE
Check compiler plugins for version conflicts and cleanup

### PR DESCRIPTION
**Pull request by sschaef**
Supersedes #925

The only unresolved thing I see for now is that `pluginsDir` is by default set to `-Xpluginsdir = misc/scala-devel/plugins`. I don't know if this directory exists at all, but if not it is filtered out in method `validateCompilerPlugins()`.

### CODE REVIEW

**Comment by dragos** [comment]
Not sure what's going on... I'll have to check it tomorrow

**Comment by dragos** [comment]
retest this please

**Comment by dragos** [comment]
retest this please

**Comment by dragos** [comment]
retest this please

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala [hunk 1]
```diff
@@ -466,32 +476,54 @@ trait ClasspathManagement extends HasLogger { self: ScalaProject =>
...
-        case VersionInFile(version) =>
-          if (!ScalaPlugin().isCompatibleVersion(version, this)) {
-            badEntries += ((entry,version))
-            val msg = s"${entry.lastSegment()} of ${this.underlying.getName()} build path is cross-compiled with an incompatible version of Scala (${version.unparse}). In case this report is mistaken, this check can be disabled in the compiler preference page."
-            errors += ((IMarker.SEVERITY_ERROR, msg, SdtConstants.ScalaVersionProblemMarkerId))
-          }
-        case _ =>
+  private def validateBinaryVersionsOnClasspath(): Seq[ClasspathErrorMarker] = {
+    scalaClasspath.userCp.foldLeft(List[ClasspathErrorMarker]()) {
+      case (errors, entry) ⇒ entry.lastSegment() match {
```
Comment (dragos) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala]: This refactoring misses the null-check. It was necessary, according to 9b44832551d465c7f32916a113f9d5f5b5086ed3

**Comment by dragos** [comment]
retest this please

**Comment by dragos** [comment]
Except for the `null` check, this looks good. I'd like to see a test, though.

**Comment by dragos** [comment]
retest this please

**Comment by dragos** [comment]
Sorry for these Jenkins blunders. The "build stuck" plugin has decided the build has frozen and aborting all builds after 3 minutes (it's set to be elastic, and since the last 3 builds finished very fast it thought that's how it should be). I set the timeout to 90 min now

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala (hunk 1 above)
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala]: Fixed

**Comment by sschaef** [comment]
Working on a test now

**Comment by dragos** [comment]
It's probably better to merge and add the test later than to miss the 4.1 release :)

**Comment by sschaef** [comment]
I opened a ticket for the tests: https://github.com/scala-ide/scala-ide/pull/935

**Pull request merged**